thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -892,42 +871,46 @@ public Trigger postNewTrigger(
             @Nullable TriggerOptions options)
         throws ThingIFException
     {
-        // TODO: implement me.
-        return null;
+        return postNewTriggerWithForm(form, predicate, options);
     }
 
-    private Trigger postNewTriggerWithCommands(
-            String schemaName,
-            int schemaVersion,
-            List<Action> actions,
-            Predicate predicate,
-            Target commandTarget) throws ThingIFException {
+    private Trigger postNewTriggerWithForm(
+            @NonNull TriggeredCommandForm form,
+            @NonNull Predicate predicate,
+            @Nullable TriggerOptions options)
+        throws ThingIFException
+    {
         if (this.target == null) {
             throw new IllegalStateException("Can not perform this action before onboarding");
         }
-        if (TextUtils.isEmpty(schemaName)) {
-            throw new IllegalArgumentException("schemaName is null or empty");
-        }
-        if (actions == null || actions.size() == 0) {
-            throw new IllegalArgumentException("actions is null or empty");
+        if (form == null) {
+            throw new IllegalArgumentException("form is null.");
         }
         if (predicate == null) {
```

Comment:
@rizakii

Android's annotation can not throws NullPointerException if @NonNull is declared.
Moreover, other APIs in thing-if AndroidSDK checks nullability if @NonNull is declared.

So, I think checking null for @NonNull argument is required.

What do you think?